src/ios/SentryCordova.m

```diff
@@ -8,121 +8,245 @@
-    }
-    if (event.extra[@"__sentry_dist"]) {
-        event.dist = [NSString stringWithFormat:@"%@", event.extra[@"__sentry_dist"]];
+- (void)startWithOptions:(CDVInvokedUrlCommand *)command {
+  [self.commandDelegate runInBackground:^{
+    NSDictionary *options = [command.arguments objectAtIndex:0];
+
+    SentryBeforeSendEventCallback beforeSend =
+        ^SentryEvent *(SentryEvent *event) {
+      [self setReleaseVersionDist:event];
+      return event;
+    };
+    [options setValue:beforeSend forKey:@"beforeSend"];
+
+    NSError *error = nil;
+
+    SentryOptions *sentryOptions = [[SentryOptions alloc] initWithDict:options
+                                                      didFailWithError:&error];
+
+    CDVPluginResult *result =
+        [CDVPluginResult resultWithStatus:CDVCommandStatus_OK
+                            messageAsBool:YES];
+    if (error != nil) {
+      NSLog(@"%@", error);
+      result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                   messageAsBool:NO];
+    } else {
+      [SentrySDK startWithOptionsObject:sentryOptions];
     }
-    event.sdk = @{@"name": SentryCordovaSdkName,
-                  @"version": SentryCordovaVersionString,
-                  @"integrations": @[@"sentry-cocoa"]};
+
+    [self.commandDelegate sendPluginResult:result
+                                callbackId:command.callbackId];
+  }];
 }
 
-- (void)sendEvent:(CDVInvokedUrlCommand *)command {
-    [self.commandDelegate runInBackground:^{
-        NSDictionary *jsonEvent = [command.arguments objectAtIndex:0];
-        if ([NSJSONSerialization isValidJSONObject:jsonEvent]) {
-            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:jsonEvent
-                                                               options:0
-                                                                 error:nil];
-
-            SentryEvent *sentryEvent = [[SentryEvent alloc] initWithJSON:jsonData];
-            if (SentryClient.sharedClient != nil) {
-                [SentryClient.sharedClient sendEvent:sentryEvent withCompletionHandler:^(NSError * _Nullable error) {
-                    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:YES];
-                    if (error != nil) {
-                        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsBool:NO];
-                    }
-                    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
-                }];
-            }
+- (void)setReleaseVersionDist:(SentryEvent *)event {
+  if (event.extra[@"__sentry_version"]) {
+    NSDictionary *infoDict = [[NSBundle mainBundle] infoDictionary];
+    event.releaseName =
+        [NSString stringWithFormat:@"%@-%@", infoDict[@"CFBundleIdentifier"],
+                                   event.extra[@"__sentry_version"]];
+  }
+  if (event.extra[@"__sentry_release"]) {
```

Comment:
where is this `__sentry_release` coming from?

Reply:
It's being set by the `setRelease` method that we export: https://github.com/getsentry/sentry-cordova/blob/5b634be27f5b21195a7162ec384aad1cdc2459ca/src/js/sdk.ts#L20-L36
We deprecated it on RN, we should do the same here on Cordova in a separate PR: https://github.com/getsentry/sentry-cordova/issues/196